File: src/test/java/br/com/zumbolovsky/fateapp/RedisSetup.java
package br.com.zumbolovsky.fateapp;

import org.testcontainers.containers.GenericContainer;

public class RedisSetup {
    private static final String image = "bitnami/redis:latest";
    private static final Integer port = 6379;

    private static GenericContainer<?> container() {
        return new GenericContainer<>(image).withExposedPorts(port);
    }

    public RedisSetup() {
        final GenericContainer<?> container = container();
        container.addEnv("ALLOW_EMPTY_PASSWORD", "yes");
        container.start();
        System.setProperty("spring.redis.host", container.getHost());
        System.setProperty("spring.redis.port", port.toString());
    }
}

File: src/test/kotlin/br/com/zumbolovsky/fateapp/RedisSetup.kt
package br.com.zumbolovsky.fateapp

import org.testcontainers.containers.GenericContainer

class RedisSetup {
    private val image = "bitnami/redis:latest"
    private val port = 6379

    private val container =
        GenericContainer(image)
            .withExposedPorts(port)

    private val host: String
    get() = container.host


    init {
        container.addEnv("ALLOW_EMPTY_PASSWORD", "yes")
        container.start()
        System.setProperty("spring.redis.host", host)
        System.setProperty("spring.redis.port", port.toString())
    }
}
